### ISSUE TITLE
csb, inetsim and https

### ISSUE THREAD

**Issue by mallorybobalice** [spender-sandbox/cuckoo-modified#191]
heh, hello. 

seen a couple of samples recently where maldocs call out to https://onlineservicexyz.com/stuffapi/params

The specific maldocs use an api service to precheck for the ip info being a set of 'disliked' services. 
the actual payload request for a powershell file only follows the service api check not failing.  

https service calls to sites from maldocs seem to break for either CA or mismatching dns name on the cert reason. For IE there's a workaround of wildcard cert + setting flag to ignore cert name mismatches, office doesn't have that. 
 
i've (temporarily as a workaround) imported a fake cert for api site in the cert SAN into inetsim and into the host as trusted root (yea yea yea, i really want to set up fake intermediate/root ca and import the those instead, then it's much easier to update the fake site certs on inetsim only) 

basically, how do people work around getting more details for HTTPs requests from maldocs? 
has anyone seen inetsim extensions to instead generate a cert for a subject or SAN matching the requested site (preferably with the same key and cert chain)

**Comment by seifreed** [comment]
hi,

You have any hash for do tests?

Regards
--
Marc Rivero López | @seifreed | www.ecrime.info <http://www.ecrime.info/>

**Comment by spender-sandbox** [comment]
He's most likely referring to: https://blogs.mcafee.com/mcafee-labs/macro-malware-adds-tricks-uses-maxmind-to-avoid-detection/

**Comment by doomedraven** [comment]
but it also easy to patch :)

**Comment by mallorybobalice** [comment]
Yea one of the mcf zscaler ones. 
Yes, takes 5s to jump to the sub to run the pshell fetch bit in disable interaction mode. 
Not the point. 
Let's also avoid the get a physical cuckoo set with real net topic. 
Personal issue is inetsim can't auto make certs for a requested dns name by itself (only uses static certs ), so that's a bit of a concern. To  say tomorrow they add it checking Google for my ip or some random site, I'd rather not manually cut a new cert for the new domain so the doc doesn't fail open (because on cert name to site name exception from vba engine. 

Any thoughts or workarounds?

**Comment by mallorybobalice** [comment]
: (

**Comment by mallorybobalice** [comment]
heh, well, now they switched to https://wtfismyip\.com/json

did a ca on the inetsim box , but still keen to hear if we can get inetsim to auto gen a cert with a list of CN or SAN  + current request . or perhaps just make office suppress the cert mismatching cert SAN error.
